### PR TITLE
Migrate umitools/group to nf-test

### PR DESCRIPTION
**Pull request by SPPearce**

### CODE REVIEW

File: modules/nf-core/umitools/group/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                    process.out.tsv,
+                    file(process.out.log[0][1]).readLines()[0..1],
+                    process.out.versions
+                    ).match() }
+            )
+        }
+
+    }
+
+    test("with_umi - bam") {
```
Comment (Joon-Klaps) [inline modules/nf-core/umitools/group/tests/main.nf.test:97]:
```suggestion
    test("with_umi - create only bam") {
```

File: modules/nf-core/umitools/group/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                    bam(process.out.bam[0][1]).getReadsMD5(),
+                    file(process.out.log[0][1]).readLines()[0..1],
+                    process.out.versions
+                    ).match() }
+            )
+        }
+
+    }
+
+    test("with_umi - tsv") {
```
Comment (Joon-Klaps) [inline modules/nf-core/umitools/group/tests/main.nf.test:122]:
```suggestion
    test("with_umi - create bam and tsv") {
```

File: modules/nf-core/umitools/group/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                input[2] = true
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.tsv,
```
Comment (Joon-Klaps) [inline modules/nf-core/umitools/group/tests/main.nf.test:138]:
```suggestion
                    process.out.tsv,
                    bam(process.out.bam[0][1]).getReadsMD5(),
```

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
Minor suggestions

**Comment by SPPearce** [comment]
Thanks @Joon-Klaps

**Pull request merged**